code/surface_area.py

```diff
@@ -0,0 +1,303 @@
+            'Maps to Surface': 'true',
+            'Number of Maps': '1',
+            'Map Name': 'vertex areas'
+        }
+        
+        validation_results = {}
+        
+        for field, expected_value in required_fields.items():
+            if field == 'Map Name':
+                # Special case for map name - check in the table
+                pattern = r'vertex areas'
+                if re.search(pattern, output, re.IGNORECASE):
+                    validation_results[field] = True
+                    print(f"✓ {field}: Found 'vertex areas'")
+                else:
+                    validation_results[field] = False
+                    print(f"✗ {field}: 'vertex areas' not found")
+            else:
+                # Check for field: value pattern
+                pattern = f"{field}:\\s*{re.escape(expected_value)}"
+                if re.search(pattern, output, re.IGNORECASE):
+                    validation_results[field] = True
+                    print(f"✓ {field}: {expected_value}")
+                else:
+                    validation_results[field] = False
+                    print(f"✗ {field}: Expected '{expected_value}' not found")
+        
+        # Check for numeric data fields
+        numeric_checks = {
+            'Number of Vertices': r'Number of Vertices:\s*(\d+)',
+            'Minimum': r'1\s+([\d.]+)',  # Map 1 minimum value
+            'Maximum': r'1\s+[\d.]+\s+([\d.]+)',  # Map 1 maximum value
+            'Mean': r'1\s+[\d.]+\s+[\d.]+\s+([\d.]+)',  # Map 1 mean value
+        }
+        
+        for check_name, pattern in numeric_checks.items():
+            match = re.search(pattern, output)
+            if match:
+                value = match.group(1)
+                try:
+                    float_value = float(value)
+                    if float_value > 0:
+                        validation_results[check_name] = True
+                        print(f"✓ {check_name}: {value} (valid positive value)")
+                    else:
+                        validation_results[check_name] = False
+                        print(f"✗ {check_name}: {value} (should be positive)")
+                except ValueError:
+                    validation_results[check_name] = False
+                    print(f"✗ {check_name}: {value} (not a valid number)")
+            else:
+                validation_results[check_name] = False
+                print(f"✗ {check_name}: Not found in output")
+        
+        # Overall validation result
+        all_passed = all(validation_results.values())
+        
+        if all_passed:
+            print("✓ All field validations passed")
+        else:
+            failed_checks = [k for k, v in validation_results.items() if not v]
+            print(f"✗ Failed validations: {', '.join(failed_checks)}")
+        
+        return all_passed
+        
+    except Exception as e:
+        print(f"✗ Validation failed: {e}")
+        return False
```

Comment:
Appreciate the need to validate the generated file. After looking at how and what  `wb_command -file-information` outputs, I think the validation here may be complicated to compensate. It may actually be better to use `nibabel` to load the generated gifti and verify:

1. Hemisphere, which should be noted in the metadata, matches that indicated in the filename
2. Similarly, the number of vertices, stored in the data array, matches the expected density (again in filename)